integration-tests/smoke/vrfv2plus_test.go

```diff
@@ -1364,6 +1367,211 @@ func TestVRFV2PlusWithBHS(t *testing.T) {
+		err = vrfv2plus.FundSubscriptions(
+			env,
+			fundingAmt,
+			fundingAmt,
+			linkToken,
+			vrfv2PlusContracts.CoordinatorV2Plus,
+			[]*big.Int{subID},
+		)
+		require.NoError(t, err, "error funding subs after request timeout")
+
+		// 6. no fulfilment should happen since timeout+1 seconds passed in the job
+		pendingReqExists, err := vrfv2PlusContracts.CoordinatorV2Plus.PendingRequestsExist(testcontext.Get(t), subID)
+		require.NoError(t, err, "error fetching PendingRequestsExist from coordinator")
+		require.True(t, pendingReqExists, "pendingRequest must exist since subID was underfunded till request timeout")
+
+		// 7. remove job and add new job with requestTimeout = 1 hour
+		vrfNode, exists := nodesMap[vrfcommon.VRF]
+		require.True(t, exists, "VRF Node does not exist")
+		resp, err := vrfNode.CLNode.API.DeleteJob(vrfNode.Job.Data.ID)
+		require.NoError(t, err, "error deleting job after timeout")
+		require.Equal(t, resp.StatusCode, 204)
+
+		chainID := env.EVMClient.GetChainID()
+		config.VRFv2Plus.General.VRFJobRequestTimeout = &blockchain.StrDuration{Duration: time.Duration(time.Hour * 1)}
+		g := errgroup.Group{}
+		var wg sync.WaitGroup
+		wg.Add(1)
+		g.Go(func() error {
+			defer wg.Done()
+			err := vrfv2plus.SetupVRFNode(vrfv2PlusContracts, chainID, config.VRFv2Plus.General,
+				vrfv2PlusData.PubKeyCompressed, l, vrfNode)
+			require.NoError(t, err, "error creating new job with 1 hr timeout")
+			return nil
+		})
+
+		// 8. Check if initial req in underfunded sub is fulfilled now, since it has been topped up and timeout increased
+		l.Debug().Str("reqID", initialReqRandomWordsRequestedEvent.RequestId.String()).
+			Str("subID", subID.String()).
+			Msg("Waiting for initalReqRandomWordsFulfilledEvent")
+		fmt.Printf("Waiting for reqID: %v, subID: %v\n", initialReqRandomWordsRequestedEvent.RequestId, subID.String())
+		initalReqRandomWordsFulfilledEvent, err := vrfv2PlusContracts.CoordinatorV2Plus.WaitForRandomWordsFulfilledEvent(
+			[]*big.Int{subID},
+			[]*big.Int{initialReqRandomWordsRequestedEvent.RequestId},
+			configCopy.VRFv2Plus.General.RandomWordsFulfilledEventTimeout.Duration,
+		)
+		require.NoError(t, err, "error waiting for initial request RandomWordsFulfilledEvent")
+
+		require.NoError(t, err, "error waiting for fulfilment of old req")
+		require.False(t, initalReqRandomWordsFulfilledEvent.OnlyPremium, "RandomWordsFulfilled Event's `OnlyPremium` field should be false")
+		require.Equal(t, isNativeBilling, initalReqRandomWordsFulfilledEvent.NativePayment, "RandomWordsFulfilled Event's `NativePayment` field should be false")
+		require.True(t, initalReqRandomWordsFulfilledEvent.Success, "RandomWordsFulfilled Event's `Success` field should be true")
+
+		wg.Wait()
+		jobRuns, err := nodesMap[vrfcommon.VRF].CLNode.API.MustReadRunsByJob(vrfNode.Job.Data.ID)
+		require.NoError(t, err, "error reading job runs")
+		require.Equal(t, len(jobRunsBeforeTest.Data)+1, len(jobRuns.Data))
```

Comment:
we can remove this, I will test the Job Runs in a separate test

Reply:
Removed